### ISSUE TITLE
Invert screen on 18220

### ISSUE THREAD

**Issue by pgaskin** [pgaskin/NickelMenu#111]

**Comment by mu22le** [comment]
I imagine you are referring to the fact that if you use the option:
nickel_setting:toggle:invert

The actual screen colors are not inverted until the next full redraw. I think one way to fix this is adding an option to force a complete redraw (that one could chain to the invert command); is such a hook available in nickel? 

The other, less ideal, option may be to change refresh rate in a way similar to what #103 suggests.

**Comment by c02y** [comment]
I don't know what `full redraw` exactly means. IMHO, toggle on and off the menu/tool bar is a full redraw, but it doesn't work as you said.

I found this issue about a month ago, it was my first time to start using NickelMenu, and I found a workaround
```
menu_item :reader :Invert Screen  :nickel_orientation  :swap 
  chain_always                    :nickel_setting      :toggle    :invert
```
If will invert the screen immediately, but it will swap the orientation, you have to manually swap the orientation back.

And don't know why the following chains do not work at all, I have to manually swap orientation twice to invert the screen.
```
menu_item :reader :Invert Screen  :nickel_orientation  :swap 
  chain_always                    :nickel_setting      :toggle    :invert
  chain_always                    :nickel_oritentation :portrait
```
or 
```
menu_item :reader :Invert Screen  :nickel_setting     :toggle    :invert
  chain_always                    :nickel_orientation :landscape
  chain_always                    :nickel_orientation :portrait   
```
or 
```
menu_item :reader :Invert Screen  :nickel_orientation :landscape
  chain_always                    :nickel_setting     :toggle    :invert
  chain_always                    :nickel_orientation :portrait

```

**Comment by mu22le** [comment]
Thanks for the hint @c02y! 

Did you test the following combination?
```
menu_item :reader :Invert Screen  :nickel_orientation  :swap 
  chain_always                    :nickel_setting      :toggle    :invert
  chain_always                    :nickel_orientation  :swap
```

It seems like it may work... I don't have my kobo with me now, but I'll give it a try this evening.

**Comment by c02y** [comment]
The doc says
```
nickel_orientation - one of:
    invert             (4.13.12638+) - Toggles between inverted/non-inverted (preserves side) (this will not work if used in a chain with swap)
    swap               (4.13.12638+) - Toggles between portrait/landscape (preserves inversion) (this will not work if used in a chain with invert)
```

So `chain_always                    :nickel_orientation  :swap` shouldn't work, and yes, I tried it too, it doesn't work either.

**Comment by mu22le** [comment]
I think the documentation is trying to say that `nickel_orientation:swap` is incompatible with `nickel_orientation:invert`, not with `nickel_setting:toggle:invert`; but if it does not work, it does not work...

Let's hope @pgaskin comes up with something.

**Comment by c02y** [comment]
> I think the documentation is trying to say that `nickel_orientation:swap` is incompatible with `nickel_orientation:invert`, not with `nickel_setting:toggle:invert`; but if it does not work, it does not work...

Yeah, I believe I misunderstood, and yes, I tried it, the invert works immediately, but the second chain_always swap line doesn't work at all, you have to swap back manually.

**Comment by roadbloack** [comment]
```
menu_item :main     :Dark Mode          :nickel_orientation  :swap 
                     chain_always       :nickel_setting      :toggle    :invert
					 chain_always       :cmd_spawn           :sleep 2
                     chain_always       :nickel_orientation  :swap
```
seems to work for me, there is the toast that spawns because of the `sleep 2`, but adding `quiet` makes it not work anymore?
anyways, i dont mind the toast, hope this helps :)
